### PR TITLE
Fixed NetworkX 3 compatibility and switched to sparse arrays (not matrices)

### LINKED ISSUE #961: Update package to use new scipy sparse *array* standard
Screen Shot 2022-06-24 at 12 38 20 PM

### PR DESCRIPTION
**Pull request by bdpedigo**
- [x] Does this PR have a descriptive title that could go in our release notes?
- [ ] Does this PR add any new dependencies?
- [x] Does this PR modify any existing APIs?
   - [ ] Is the change to the API backwards compatible?
     - no
- [ ] Have you built the documentation (reference and/or tutorial) and verified the generated documentation is appropriate?

#### Reference Issues/PRs
Fixes #961
Proper fix of #1015 

#### Any other comments?
- Technically, we're losing support for `nx.OrderedGraph`, but I'm not losing and sleep over that one. I was even considering just switching to NetworkX 3 support only but I don't think it's fully necessary.

### CODE REVIEW

File: graspologic/pipeline/embed/_types.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from typing import Union
```
Comment (tathey1) [inline graspologic/pipeline/embed/_types.py:1]: cute

File: graspologic/utils/utils.py [hunk 2]
```diff
@@ -360,7 +367,11 @@ def remove_loops(graph: GraphRepresentation) -> Union[np.ndarray, csr_matrix]:
     """
     graph = import_graph(graph)
 
-    dia = diags(graph.diagonal()) if isspmatrix_csr(graph) else np.diag(np.diag(graph))
+    dia = (
```
Comment (tathey1) [inline graspologic/utils/utils.py:370]: what is this doing? if its already a csr_array you make it a csr_array?

**Review by tathey1 — COMMENTED** [review]
There was one line I didn't immediately understand.

Out of curiosity, how did you find and iterate through all the instances of "csr_matrix"?

File: graspologic/utils/utils.py (hunk 2 above)
Comment (bdpedigo) [reply graspologic/utils/utils.py:370]: scipy still by default returns a csr_matrix but we need a csr_array

**Comment by bdpedigo** [comment]
> 

vs code has a "rename symbol" command that can do it all in one go

**Pull request merged**